### ISSUE TITLE
[Bug]:  Navbar Dropdown Menu unreadable

### ISSUE THREAD

**Issue by peschmae** [materializecss/materialize#410]
### Before submitting...

- [X] I have searched for [duplicate or closed issues](https://github.com/materializecss/materialize/issues?utf8=%E2%9C%93&q=is%3Aissue).
- [X] I have read the [CONTRIBUTING document](https://github.com/materializecss/materialize/blob/master/CONTRIBUTING.md) and my issue is following the guidelines.
- [X] I have read the template completely before filling it in.

### Context

Opening a dropdown in the navbar menu opens properly, but the menu entries are unreadable due to the styling.

### Current Behavior

Font-color for the navbar dropdown menu aren't respected

### Expected behavior

In earlier versions the font color and styling in the navbar menu matched the other dropdowns

### Possible Solutions or Causes

Looking at the applied CSS, shows that the following code is applied, and I assume the CSS selector is somehow wrong
```css
nav ul:not(.dropdown-content) a {
    -webkit-transition: background-color .3s;
    transition: background-color .3s;
    font-size: 1rem;
    color: var(--font-on-primary-color-main);
    display: block;
    padding: 0 15px;
    cursor: pointer
}
```
This code was introduced for #254 to introduce the dark theme mentioned in #77 

The testbed used for the pull request (https://materialize-testbed.s3-us-west-2.amazonaws.com/dark-theme/index.html# ) doesn't have a navbar dropdown, so it can't be reproduced there

### Steps to reproduce

Reproducable on the official documentation at https://materializeweb.com/navbar.html#navbar-dropdown
Copying the example code from there to codepen shows the same issue.

### Your Environment

* Version used: 2.0.2-alpha
* Browser Name and version: latest chrome, latest firefox, latest Safari
* Operating System and version (desktop or mobile): macOS Ventura 13.4.1
* Additional information you want to tell us: no codepen supplied as it can be reproduced the example code from the official documentation

**Comment by wuda-io** [comment]
Hey @peschmae 
thats right, thanks for pointing that out! I also noticed, that the hovering does only hover the text.

![grafik](https://github.com/materializecss/materialize/assets/27806175/a0e3aa96-2400-4d82-8900-72870fe9eaaf)

Does your solution also fix that? Thanks for your contribution!

**Comment by peschmae** [comment]
Hi @wuda-io 
Yes, the hover is also fixed with this change.

**Comment by wuda-io** [comment]
Merged with #411
